index.js

```diff
@@ -15,9 +15,23 @@ module.exports = async function serve (drive, opts = {}) {
       return
     }
 
-    const { pathname } = new URL(req.url, 'http://localhost')
+    const { pathname, searchParams } = new URL(req.url, 'http://localhost')
+    const version = searchParams.get('checkout')
+    const snapshot = version ? drive.checkout(version) : drive
```

Comment:
Just realized `localdrive` doesn't have a `checkout` method, and it could be deceiving for localdrive to have it, so maybe just this:
`const snapshot = version && drive.checkout ? drive.checkout(version) : drive`

wdyt?

Reply:
Otherwise I could just add it to localdrive, maybe it ends up being good for more compatibility

Reply:
Good catch, I'll get back to it tomorrow.
Initial thought: it depends on what we mean with 'File system interoperable with Hyperdrive' on the Localdrive readme--the deceiving methods could also be implemented as reasonable no-ops, for example returning the localdrive unchanged for checkout (I think I prefer this approach, because then we can use a localdrive in any code where a hyperdrive is used)

Reply:
I'll expand the tests tomorrow to work with both hyperdrive and localdrive, since this code should work for both

Reply:
Localdrive tests added. The one requiring checkout is failing pending this PR: https://github.com/holepunchto/localdrive/pull/20